From 3bb1addeec1f68232dfbe2c373a22b7596ec7841 Mon Sep 17 00:00:00 2001
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Thu, 14 Feb 2019 17:38:24 +0100
Subject: [PATCH 027/252] serial: 8250: remove that trylock in
 serial8250_console_write_atomic()

This does not work as rtmutex in NMI context. As per John, it is not
needed.

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
---
 drivers/tty/serial/8250/8250_port.c | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/drivers/tty/serial/8250/8250_port.c b/drivers/tty/serial/8250/8250_port.c
index e6fee74dff2e..8f8c925d7045 100644
--- a/drivers/tty/serial/8250/8250_port.c
+++ b/drivers/tty/serial/8250/8250_port.c
@@ -3275,17 +3275,9 @@ void serial8250_console_write_atomic(struct uart_8250_port *up,
 {
 	struct uart_port *port = &up->port;
 	unsigned int flags;
-	bool locked;
 
 	console_atomic_lock(&flags);
 
-	/*
-	 * If possible, keep any other CPUs from working with the
-	 * UART until the atomic message is completed. This helps
-	 * to keep the output more orderly.
-	 */
-	locked = spin_trylock(&port->lock);
-
 	touch_nmi_watchdog();
 
 	clear_ier(up);
@@ -3300,9 +3292,6 @@ void serial8250_console_write_atomic(struct uart_8250_port *up,
 	wait_for_xmitr(up, BOTH_EMPTY);
 	restore_ier(up);
 
-	if (locked)
-		spin_unlock(&port->lock);
-
 	console_atomic_unlock(flags);
 }
 
-- 
2.19.1

